### PR TITLE
fix: native share text concat and comment addition

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Added a concatenated version of the text including text + url
- Also added the native share feature to the comments

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-272 #done

### CODE REVIEW

File: packages/shared/src/components/post/PostWidgets.tsx [hunk 1]
```diff
@@ -35,8 +35,7 @@ export function PostWidgets({
     if ('share' in navigator) {
       try {
         await navigator.share({
-          text: post.title,
-          url: post.commentsPermalink,
+          text: `${post.title} ${post.commentsPermalink}`,
```
Comment (idoshamun) [inline packages/shared/src/components/post/PostWidgets.tsx:38]: Not sure if this is preferred. Just a suggestion

```suggestion
          text: `${post.title}\n${post.commentsPermalink}`,
```

File: packages/shared/src/hooks/useShareComment.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  closeShareComment: () => void;
+} {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [shareModal, setShareModal] = useState<Comment>(null);
+
+  const openShareComment = async (comment: Comment) => {
+    if ('share' in navigator) {
+      try {
+        await navigator.share({
+          text: `${post.title} ${post.commentsPermalink}${getCommentHash(
```
Comment (idoshamun) [inline packages/shared/src/hooks/useShareComment.ts:24]: Same suggestion can be applied here

File: packages/shared/src/hooks/useShareComment.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    } else {
+      setShareModal(comment);
+    }
+  };
+
+  const closeShareComment = () => {
+    setShareModal(null);
+  };
+
+  return {
```
Comment (idoshamun) [inline packages/shared/src/hooks/useShareComment.ts:49]: You should use `useMemo` here if possible and define the functions inside

**Review by idoshamun — APPROVED** [review]
Some optimization suggestions. Other than that, it looks great

**Comment by rebelchris** [comment]
@idoshamun Thanks for that!
always the useMemo 😅

Anyway added it to both new hooks and added the new lines

**Comment by rebelchris** [comment]
@idoshamun Thanks a lot!
Learned a lot from this PR

**Pull request merged**